rancher/rancher
# Fix the issue where Rancher server startup gets stuck if KDM is blocked

## Issue: 

 
https://github.com/rancher/rancher/issues/40416

## Problem

 
Rancher server hangs when it can not get the response from the KDM URL (https://github.com/rancher/kontainer-driver-metadata). 

The root cause of this problem is that the appropriate HTTP request timeout is not set in the channelserver lib. The channelserver waits for the response and never exists. 

## Solution

 
The fix is to add an appropriate timeout to the HTTP request in the channelserver. ([PR](https://github.com/rancher/channelserver/pull/23))

## Testing


## Engineering Testing
### Manual Testing


The PR has been dev-validated in my local environment.
By following https://github.com/rancher/rancher/issues/40416, I can confirm that rancher server is not stuck in waiting for the response from the KDM URL.


### Automated Testing

* Test types added/modified: None
* If "None" - Reason:  

[The way the channel server is used in rancher ](https://github.com/rancher/rancher/blob/3a80de7242a61eea67be22bca7c83763a1402b5a/pkg/channelserver/channelserver.go#L112-L124)makes it impossible to write a unit test because I can not manipulate the KDM URL in the unit test. 

Also, it is too involved to write an e2e test for this small change. 

  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

 
## QA Testing Considerations

 
Please follow the steps in the GH issue to prepare a web server and use it when running Rancher. 
Make sure Rancher serve does not hang. 

### Regressions Considerations


N/A 

Existing / newly added automated tests that provide evidence there are no regressions: N/A
